src/platform/imx8m/lib/dma.c

```diff
@@ -12,6 +12,7 @@
 #include <sof/spinlock.h>
 
 extern struct dma_ops dummy_dma_ops;
+extern struct dma_ops sdma_ops;
 
```

Comment:
@paulstelian97 can we remove do not merge and add a comment about what needs to be done? In this way we will have booting with qemu ready and we fix the rest later.

Reply:
Alright. Mentioned in the commit message that a way to account for channel 0 is needed. As for our functional tests not succeeding, we will still need to investigate (perhaps open a GH issue to keep track of it)

Reply:
@paulstelian97 there are still some checkpatch.pl issues, care to have a look?

Reply:
Resolved most of what I could spot, 2 cannot be solved without significant changes. I need to figure out a way to reduce the indentation in those two cases.

Reply:
@paulstelian97 thanks Paul. Looks good to me now. Will merge this change now after CI does the basic checks.

Reply:
Fixed more of them as I noticed them in CI.